components/region_cache_memory_engine/src/engine.rs

```diff
@@ -357,12 +375,20 @@ impl Mutable for RegionCacheWriteBatch {
 pub struct RegionCacheSnapshot {
     region_id: u64,
     snapshot_ts: u64,
+    // Sequence number is shared between RegionCacheEngine and disk KvEnigne to
+    // provide atomic write
+    sequence_number: u64,
```

Comment:
When would `squence_number` and `snapshot_ts` differ? Doesn't affect my iteration change, since I am accessing the sequence number by calling .sequence_number(), so my code should be fine.

Reply:
Actually answered my question: the sequence number would be different from snapshot_ts as we write with one ts, but read with another. The write ts is provided by the RocksDb engine, we need to get it from the write callback when rocksdb writes.Reads should be up to the snapshot. I think this will be more clear once I implement batching.

Reply:
Yes, they are totally different things. snapshot_ts is acutually the mvcc version and is not used in read at the level of HybridEngineIterator. It is here to not let data that is being visible for some snapshots not removed by GC and make safe _ts not go beyound it.